hub-saml/src/main/java/uk/gov/ida/saml/hub/transformers/outbound/OutboundResponseFromHubToSamlResponseTransformer.java

```diff
@@ -27,9 +31,9 @@ public OutboundResponseFromHubToSamlResponseTransformer(
     @Override
     protected void transformAssertions(OutboundResponseFromHub originalResponse, Response transformedResponse) {
         originalResponse
-                .getMatchingServiceAssertion()
+                .getEncryptedAssertions().stream()
                 .map(encryptedAssertionUnmarshaller::transform)
-                .map(transformedResponse.getEncryptedAssertions()::add);
+                .forEach(transformedResponse.getEncryptedAssertions()::add);
```

Comment:
Not only is forEach() more idiomatic here, it turns out map() won't work if the source stream came from a List rather than an Optional, so this change was crucial.  No idea why that's the case.

Reply:
I have been learning a little about Streams, don't know much but it could be that it because map is a non-terminal stream operator while foreach is Terminal

`
originalResponse
                .getEncryptedAssertions().stream()
                .map(encryptedAssertionUnmarshaller::transform)
                .map(transformedResponse.getEncryptedAssertions()::add)
                .collect(Collectors.toList());
`
The above also seems to work with map, not sure why map worked with Optional. Perhaps Andy would know.